lib/train/transports/azure.rb

```diff
@@ -0,0 +1,108 @@
+# encoding: utf-8
+
+require 'train/plugins'
+require 'ms_rest_azure'
+require 'azure_mgmt_resources'
+require 'inifile'
+
+module Train::Transports
+  class Azure < Train.plugin(1)
+    name 'azure'
+    option :tenant_id, default: ENV['AZURE_TENANT_ID']
+    option :client_id, default: ENV['AZURE_CLIENT_ID']
+    option :client_secret, default: ENV['AZURE_CLIENT_SECRET']
+    option :subscription_id, default: ENV['AZURE_SUBSCRIPTION_ID']
+
+    # This can provide the client id and secret
+    option :credentials_file, required: true, default: ENV['AZURE_CRED_FILE']
```

Comment:
Seems weird to mark this required but also provide a default. Is this actually a required parameter? Looks like we do some magic later in a separate method when parsing the creds file to assume a reasonable default file location.